ipatests/pytest_plugins/integration/__init__.py

```diff
@@ -179,8 +183,6 @@ def collect_logs(name, logs_dict, logfile_dir=None, beakerlib_plugin=None):
             else:
                 shutil.rmtree(topdirname)
 
-        logs_dict.clear()
```

Comment:
Your Jedi mind tricks ("the dict of logs to be collected should not be cleared") don't work on me. What was the original purpose of this clear? Why does not not apply now?

Reply:
Let me try to explain it.

To be honest, I do know what was the original purpose of the clear. The commit 29c28786e35ee8e7fced740e62e0b5ddaa9bb381 that introduces the code, does not explain it.

First, is important to know that:
- The `mh` fixture is called only once per class and it is there where we add data to the `logs_dict`. 
- The `integration_logs` is called in every test method.

So, running the code step by step: 
1. in the very first time, the `integration_logs` fixture is called and the `class_integration_log` is empty. It only does `yield class_integration_log` (an empty dict for now).
2. After that, the code goes to the `mh` fixture. [Here](https://github.com/felipevolpone/freeipa/blob/master/ipatests/pytest_plugins/integration/__init__.py#L239-L248) is where the name of the logs that should be collected are added. Some lines below, it calls `yield mh.install()` which start the process of installing ipa server and etc.
3. After the yield above, the flow goes back to the `integration_logs` fixture with the `class_integration_log` having all the logs to be collected. This happens because the `class_integration_log` and the `mh` fixture share the same object via parameter. They're pointing to the same object.
4. In `integration_logs` we call collect_test_logs` which expects to have the `logfile_dir` option to actually collect the logs described in the `logs_dict`.

And that's why we should not clear the dict. It is only built once per class, based on the topology.

I tried to make it more obvious, but this part of the code is quite confusing. If it's not clear, please, let me know.